File: lihua-vue/src/api/system/role/type/sys-role.ts
export interface SysRole {
  /**
   * 主键
   */
  id?: string;
  /**
   * 角色名称
   */
  name?: string;
  /**
   * 角色编码
   */
  code?: string;
  /**
   * 角色状态
   */
  status?: string;
  /**
   * 备注
   */
  remark?: string;
  /**
   * 菜单id集合
   */
  menuIds?: string[] | { checked: string[] };
}

export interface SysRoleVO extends SysRole{
  statusIsNormal?: boolean,
  updateStatusLoading?: boolean
}


export interface SysRoleDTO {
  /**
   * 角色名称
   */
  name: string;
  /**
   * 角色编码
   */
  code: string;
  /**
   * 角色状态
   */
  status: string | null;
  /**
   * 当前页数
   */
  pageNum: number;
  /**
   * 每页记录数
   */
  pageSize: number;
}

File: lihua-biz/lihua-system/src/main/java/com/lihua/system/entity/SysRole.java
package com.lihua.system.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.lihua.mybatis.model.BaseEntity;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.util.ArrayList;
import java.util.List;

@EqualsAndHashCode(callSuper = true)
@Data
public class SysRole extends BaseEntity {
    /**
     * 主键
     */
    @TableId(type = IdType.ASSIGN_ID)
    private String id;

    /**
     * 角色名称
     */
    @NotNull(message = "请输入角色名称")
    @Size(max = 60, message = "角色名称长度不能超过60个字符")
    private String name;

    /**
     * 角色编码
     */
    @NotNull(message = "请输入角色编码")
    @Size(max = 100, message = "角色编码长度不能超过100个字符")
    private String code;

    /**
     * 角色状态
     */
    @NotNull(message = "请选择角色状态")
    @Pattern(regexp = "^[01]$", message = "角色状态不合法")
    private String status;

    /**
     * 备注
     */
    @Size(max = 500, message = "备注长度不能超过500个字符")
    private String remark;

    /**
     * 菜单id集合
     */
    @TableField(exist = false)
    private List<String> menuIds = new ArrayList<>();
}
